utils/azureCredentialsManager.ts

```diff
@@ -0,0 +1,145 @@
+        let registries: ContainerModels.Registry[] = [];
+
+        if (subscription && resourceGroup) {
+            //Get all registries under one resourcegroup
+            const client = this.getContainerRegistryManagementClient(subscription);
+            registries = await client.registries.listByResourceGroup(resourceGroup);
+
+        } else if (subscription) {
+            //Get all registries under one subscription
+            const client = this.getContainerRegistryManagementClient(subscription);
+            registries = await client.registries.list();
+
+        } else {
+            //Get all registries for all subscriptions
+            const subs: SubscriptionModels.Subscription[] = this.getFilteredSubscriptionList();
+            const subPool = new AsyncPool(MAX_CONCURRENT_SUBSCRIPTON_REQUESTS);
+
+            for (let i = 0; i < subs.length; i++) {
+                subPool.addTask(async () => {
+                    const client = new ContainerRegistryManagementClient(this.getCredentialByTenantId(subs[i].tenantId), subs[i].subscriptionId);
+                    let registry: ContainerModels.Registry[] = await client.registries.list();
+                    registries = registries.concat(registry);
+                });
+            }
+            await subPool.runAll();
+        }
+
+        if (sortFunction) {
+            registries.sort(sortFunction);
+        }
+
+        return registries;
+    }
+
+    public async getResourceGroups(subscription?: SubscriptionModels.Subscription): Promise<ResourceGroup[]> {
+        if (subscription) {
+            const resourceClient = new ResourceManagementClient(this.getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+            return await resourceClient.resourceGroups.list();
+        }
+        const subs = this.getFilteredSubscriptionList();
+        const subPool = new AsyncPool(MAX_CONCURRENT_SUBSCRIPTON_REQUESTS);
+        let resourceGroups: ResourceGroup[] = [];
+        //Acquire each subscription's data simultaneously
+        for (let i = 0; i < subs.length; i++) {
+            subscription = subs[i];
+            subPool.addTask(async () => {
+                const resourceClient = new ResourceManagementClient(this.getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+                const internalGroups = await resourceClient.resourceGroups.list();
+                resourceGroups = resourceGroups.concat(internalGroups);
+            });
+        }
+        await subPool.runAll();
+        return resourceGroups;
+    }
+
+    public getCredentialByTenantId(tenantId: string): ServiceClientCredentials {
+
+        const session = this.azureAccount.sessions.find((s, i, array) => s.tenantId.toLowerCase() === tenantId.toLowerCase());
+
+        if (session) {
+            return session.credentials;
+        }
+
+        throw new Error(`Failed to get credentials, tenant ${tenantId} not found.`);
+    }
+
+    //ERR CHECKS
```

Comment:
What is the code for //ERR CHECKS?

Reply:
/ERR CHECKS is meant to be a section whose only member is isLoggedIn()

Reply:
I see. I just feel a bit of odd to categorize login check to err check.

Reply:
Ive just put it under CHECKS which seems more reasonable